### PR TITLE
Assorted fixes

### LINKED ISSUE #3495: Update AMD OpenCL docs
This has not been applicable since the release of amdgpu-pro. It needs to be updated.

```
==== If you have an AMD GPU (or want to use CPU as OpenCL device)

- Up to Ubuntu 15.10 (<= 15.10). This will work on Intel hardware but warnings will be seen.
sudo apt-get install ocl-icd-opencl-dev opencl-headers fglrx-dev

- Ubuntu Xenial 16.04 LTS and above (>= 16.04)
The fglrx driver would not be supported in 16.04 (XServer 1.18 is not supported).
Canonical and AMD decided to use the new hybrid driver stack for 16.04. JtR OpenCL
users should stay on a supported release until the hybrid stack is available.
```

BTW: it seems CPU support no longer exits.

### PR DESCRIPTION
**Pull request by claudioandre-br**
Basically, set GWS to next LWS multiple larger than count for "my" formats.

Closes #3495

### CODE REVIEW

**Comment by claudioandre-br** [comment]
This code (that enforces 128) is 4 years old. See  https://github.com/magnumripper/JohnTheRipper/commit/f5631161ae6362a6033bd6f4e340a1d2a0751745

1. the commits above are, basically, a `git revert`;
2. so, I prefer to keep the commits as is (and create a new one to use LWS=NULL);

I don't know for sure, but off the top of my head, there is a reason (at least, there was); related to GPU mask mode
I need to stress  it to remember.

**Comment by claudioandre-br** [comment]
I'll migrate to circleCI.com. My Circle run is no longer connected to GitHub PRs.

No wait, I'm already on whateverCI.com (https://circleci.com/gh/claudioandre-br/JohnTheRipper/1352). I'll take a look.

**Comment by claudioandre-br** [comment]
> For the raw formats you break that now and enforce 128 instead - any special reason?

Wait a second. What happens if one is using `__local` memory? What happens in a call of `get_local_id(0)`? I will look at it closely later, but, I'm confident I had a reason.

File: src/opencl_rawsha256_fmt_plug.c [hunk 1]
```diff
@@ -607,9 +607,10 @@ static int crypt_all(int *pcount, struct db_salt *_salt)
 {
 	const int count = *pcount;
 	const struct db_salt *salt = _salt;
-	size_t gws = count;
-	size_t *lws = (local_work_size && !(gws % local_work_size)) ?
-		&local_work_size : NULL;
+	size_t gws, initial = 128;
+	size_t *lws = local_work_size ? &local_work_size : &initial;
+
+	gws = GET_MULTIPLE_OR_BIGGER(count, local_work_size);
```
Comment (claudioandre-br) [inline src/opencl_rawsha256_fmt_plug.c]: It seems I picked a bad name. I will propose a patch to rename it do `GET_NEXT_MULTIPLE`.

**Comment by magnumripper** [comment]
> > For the raw formats you break that now and enforce 128 instead - any special reason?
> 
> Wait a second. What happens if one is using `__local` memory? What happens in a call of `get_local_id(0)`? I will look at it closely later, but, I'm confident I had a reason.

Ah yes, that's a reason, I think my RC4 formats also need to enforce some LWS limits. Anyway, I'll approve this and we can change things later if at all needed.

**Pull request merged**